components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -2012,6 +2017,40 @@ public WSDLArchiveInfo extractAndValidateWSDLArchive(InputStream inputStream)
         return archiveInfo;
     }
 
+    @Override
+    public void sendMailNotification(APIPublisher apiPublisher, String apiId, String apiVersion, String apiName)
```

Comment:
Any reason for passing the APIPublisher here? I think we can pass the subscription list here.

Reply:
APIPublisher uses to get the subscription list.  so is it ok to get the subscription list inside from that rest API, I thought it also some business logic

Reply:
Use some other method to get the subscription list within core bundle. As I mentioned earlier, sendMailNotification method should be part of createNewAPIVersion